src/App.tsx

```diff
@@ -127,6 +127,21 @@ const Root: FunctionComponent = () => {
 	const [t] = useTranslation();
 	const [isUnsavedChangesModalOpen, setIsUnsavedChangesModalOpen] = useState(false);
 
+	useEffect(() => {
+		const url = new URL(window.location.href);
+
+		// Use one-shot function for type safety
+		((key: Avo.Auth.IdpLinkedSuccessQueryParam = 'linked') => {
+			const hasLinked = url.searchParams.get(key) !== null;
+
+			if (hasLinked) {
+				ToastService.success(t('app___je-account-is-gekoppeld'));
+				url.searchParams.delete(key);
+				history.replace(url.toString().replace(url.origin, ''));
+			}
+		})();
+	}, []);
```

Comment:
This is a strange way to enforce a type
Why not just define it as a typed variable, then the typescript compiler will complain if it isn't the correct type:
![image](https://user-images.githubusercontent.com/1710840/193528603-2b9dbcbb-9192-44fa-b2c8-ce26fc805f95.png)

```suggestion
	useEffect(() => {
		const url = new URL(window.location.href);
		const linked: Avo.Auth.IdpLinkedSuccessQueryParam = 'linked';
		const hasLinked = url.searchParams.get(linked) !== null;
		if (hasLinked) {
			ToastService.success(t('app___je-account-is-gekoppeld'));
			url.searchParams.delete(linked);
			history.replace(url.toString().replace(url.origin, ''));
		}
	}, []);
```